Taskfile.yml

```diff
@@ -67,9 +67,9 @@ tasks:
     deps: 
     - generator:ci
     - controller:ci
-    - asoctl:ci
     - crossplane:ci
     cmds:
+    - task: asoctl:ci
```

Comment:
Im not convinced that this is the issue. If you look at an example stuck run: 
https://github.com/Azure/azure-service-operator/actions/runs/4249026647/jobs/7388790440

The final logs are:
```
[generator:unit-tests-cover] go: downloading golang.org/x/text v0.7.0
[generator:unit-tests-cover] go: downloading github.com/josharian/intern v1.0.0
[generator:unit-tests-cover] go: downloading golang.org/x/sys v0.5.0
task: [controller:generate-genruntime-deepcopy] find ./pkg/genruntime -type f -name "zz_generated.*" -delete
task: [controller:generate-genruntime-deepcopy] cd pkg/genruntime && controller-gen object:headerFile=/workspace/v2/boilerplate.go.txt paths=./...
task: [controller:generate-genruntime-deepcopy] cd pkg/genruntime && gofmt -l -s -w .
[controller:generate-genruntime-deepcopy] conditions/zz_generated.deepcopy.go
[controller:generate-genruntime-deepcopy] zz_generated.deepcopy.go
Error: The operation was canceled.
```

To me that looks like either `controller:generate-genruntime-deepcopy` or `generator:unit-tests-cover` are getting stuck, not `asoctl`, which can't even be building at the time those steps run as it has a dependency on `controller:generate-crds` which hasn't happened yet.

I think we should add some logs to the end of the  `controller:generate-genruntime-deepcopy` and `generator:unit-tests-cover` steps to see if they are actually finishing. Then when things get stuck again we can determine if the issue is either of those commands.

We may also want to stick a log at the beginning of `produce-markdown-summary` as that task is deferred til after the `generator:unit-tests-cover` finishes -- so it should run immediately after, but obviously isn't.